fix: LEAP-1004: Show prediction with selected project.model_version in QuickView

niklub opened the pull request:
In "QuickView" mode, user now sees the prelabeled data from the prediction based on selected model version.
If prediction from this model is not presented it fallbacks to old behavior and uses the first prediction.

**Model version selected**:
<img width="915" alt="image" src="https://github.com/HumanSignal/label-studio/assets/6087484/dcdb86db-bd7f-4476-ac98-724ecd991201">

**Correct prediction is shown first**:
<img width="1437" alt="image" src="https://github.com/HumanSignal/label-studio/assets/6087484/2e617406-a4c1-4f14-a774-077469810a00">
<img width="1420" alt="image" src="https://github.com/HumanSignal/label-studio/assets/6087484/edc2ebdb-923d-4fbc-88ea-745254da2106">


### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)


#### Change has impacts in these area(s)
- [x] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

Commit:
fix: LEAP-1004: Show prediction with select project.model_version in QuickView

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8688487227

niklub committed:
Update lsf-sdk.js

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8688821147

deppp commented:
Why do we think this is a correct behavior? That should be an EXPLICIT user action to select what predictions should be used?

niklub commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8705834891)
> Successfully merged:  327 files changed, 70 insertions(+), 14 deletions(-)

niklub committed:
Merge branch 'develop' into 'fb-LEAP-1004'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8705834891

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8705865912

niklub commented:
> Why do we think this is a correct behavior? That should be an EXPLICIT user action to select what predictions should be used?

In the current implementation, the only explicit user action is to set the model version in project settings, which produces the bug fixed in this PR.

Two alternatives that could be better from UX perspective:
- either let the user create annotation from prediction manually, without pre-initializing it (downside: it can break existing QuickView labeling flow without or with a single prediction, or introduce inconsistency)
- or let the user copy predictions into the draft annotation (downside: this tool has to be implemented)

Is there any other scenario that can better address the problem?

Anyway, this PR obviously fixes an incorrect behavior, and we can address future enhancements in the next ones
